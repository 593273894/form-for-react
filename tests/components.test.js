import React from 'react';
import { Form, Input, RadioGroup, CheckboxGroup, Select } from './form';

test('form component change', () => {
    let name;
    let value;
    const component = mount(
        <Form>
            <Input
                name="input"
                label="input"
                onChange={(paraName, paraValue) => { name = paraName; value = paraValue }} />
            <RadioGroup
                name="radio"
                label="radio"
                options={[{
                    label: 'red', value: 'red'
                }, {
                    label: 'yellow', value: 'yellow'
                }, {
                    label: 'blue', value: 'blue'
                }]}
                onChange={(paraName, paraValue) => { name = paraName; value = paraValue }} />
            <CheckboxGroup
                name="checkbox"
                label="checkbox"
                options={[{
                    label: 'apple', value: 'apple'
                }, {
                    label: 'banana', value: 'banana'
                }]}
                onChange={(paraName, paraValue) => { name = paraName; value = paraValue }} />
            <Select
                name="select"
                label="select"
                options={[{
                    label: 'cat', value: 'cat'
                }, {
                    label: 'dog', value: 'dog'
                }]}
                onChange={(paraName, paraValue) => { name = paraName; value = paraValue }} />
        </Form>
    );
    expect(component).toMatchSnapshot();

    component.find('#input').simulate('change', { target: { value: 'apple' } });
    expect(name).toBe('input');
    expect(value).toBe('apple');

    component.find('#radio').childAt(0).childAt(0).simulate('change');
    expect(name).toBe('radio');
    expect(value).toBe('red');
    component.find('#radio').childAt(1).childAt(0).simulate('change');
    expect(name).toBe('radio');
    expect(value).toBe('yellow');

    component.find('#checkbox').childAt(0).childAt(0).simulate('change');
    expect(name).toBe('checkbox');
    expect(value).toEqual(['apple']);
    component.find('#checkbox').childAt(1).childAt(0).simulate('change');
    expect(name).toBe('checkbox');
    expect(value).toEqual(['apple', 'banana']);
    component.find('#checkbox').childAt(0).childAt(0).simulate('change');
    expect(name).toBe('checkbox');
    expect(value).toEqual(['banana']);
    component.find('#checkbox').childAt(1).childAt(0).simulate('change');
    expect(name).toBe('checkbox');
    expect(value).toEqual([]);

    component.find('#select').simulate('change', { target: { value: 'cat' } });
    expect(name).toBe('select');
    expect(value).toEqual('cat');
    component.find('#select').simulate('change', { target: { value: 'dog' } });
    expect(name).toBe('select');
    expect(value).toEqual('dog');

    expect(component).toMatchSnapshot();
});

test('form submit', () => {
    let output;
    const component = mount(
        <Form onSubmit={(values) => output = values}>
            <Input
                name="input"
                label="input" />
            <RadioGroup
                name="radio"
                label="radio"
                options={[{
                    label: 'red', value: 'red'
                }, {
                    label: 'yellow', value: 'yellow'
                }, {
                    label: 'blue', value: 'blue'
                }]} />
            <CheckboxGroup
                name="checkbox"
                label="checkbox"
                options={[{
                    label: 'apple', value: 'apple'
                }, {
                    label: 'banana', value: 'banana'
                }]} />
            <Select
                name="select"
                label="select"
                options={[{
                    label: 'cat', value: 'cat'
                }, {
                    label: 'dog', value: 'dog'
                }]} />
        </Form>
    );
    expect(component).toMatchSnapshot();

    component.find('#input').simulate('change', { target: { value: 'apple' } });
    component.find('#radio').childAt(0).childAt(0).simulate('change');
    component.find('#checkbox').childAt(0).childAt(0).simulate('change');
    component.find('#select').simulate('change', { target: { value: 'cat' } });
    component.find('.form').simulate('submit');
    expect(output).toEqual({
        "input": 'apple',
        "radio": 'red',
        "checkbox": ['apple'],
        "select": 'cat',
    });

    expect(component).toMatchSnapshot();
});