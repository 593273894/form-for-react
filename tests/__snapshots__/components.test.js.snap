// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form component change 1`] = `
<Form
  onSubmit={[Function]}
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <FormField
      label="input"
      name="input"
      onChange={[Function]}
    >
      <div
        className="form-field"
      >
        <label
          className="field-label"
          htmlFor="input"
        >
          <span
            className="required"
          />
          <span
            className="label"
          >
            input
          </span>
        </label>
        <div
          className="field-input"
        >
          <Input
            api={
              Object {
                "addField": [Function],
                "changeField": [Function],
                "getErrors": [Function],
                "getTouched": [Function],
                "getValues": [Function],
                "removeField": [Function],
                "setValue": [Function],
              }
            }
            changeFieldValue={[Function]}
            label="input"
            name="input"
          >
            <input
              autoComplete="off"
              className="input"
              id="input"
              onChange={[Function]}
              onFocus={[Function]}
              type="text"
              value=""
            />
          </Input>
        </div>
      </div>
    </FormField>
    <FormField
      label="radio"
      name="radio"
      onChange={[Function]}
      options={
        Array [
          Object {
            "label": "red",
            "value": "red",
          },
          Object {
            "label": "yellow",
            "value": "yellow",
          },
          Object {
            "label": "blue",
            "value": "blue",
          },
        ]
      }
    >
      <div
        className="form-field"
      >
        <label
          className="field-label"
          htmlFor="radio"
        >
          <span
            className="required"
          />
          <span
            className="label"
          >
            radio
          </span>
        </label>
        <div
          className="field-input"
        >
          <RadioGroup
            api={
              Object {
                "addField": [Function],
                "changeField": [Function],
                "getErrors": [Function],
                "getTouched": [Function],
                "getValues": [Function],
                "removeField": [Function],
                "setValue": [Function],
              }
            }
            changeFieldValue={[Function]}
            label="radio"
            name="radio"
            options={
              Array [
                Object {
                  "label": "red",
                  "value": "red",
                },
                Object {
                  "label": "yellow",
                  "value": "yellow",
                },
                Object {
                  "label": "blue",
                  "value": "blue",
                },
              ]
            }
          >
            <div
              className="radio-group"
              id="radio"
            >
              <label
                className="radio"
              >
                <input
                  checked={false}
                  className="radio-input"
                  onChange={[Function]}
                  type="radio"
                />
                red
              </label>
              <label
                className="radio"
              >
                <input
                  checked={false}
                  className="radio-input"
                  onChange={[Function]}
                  type="radio"
                />
                yellow
              </label>
              <label
                className="radio"
              >
                <input
                  checked={false}
                  className="radio-input"
                  onChange={[Function]}
                  type="radio"
                />
                blue
              </label>
            </div>
          </RadioGroup>
        </div>
      </div>
    </FormField>
    <FormField
      label="checkbox"
      name="checkbox"
      onChange={[Function]}
      options={
        Array [
          Object {
            "label": "apple",
            "value": "apple",
          },
          Object {
            "label": "banana",
            "value": "banana",
          },
        ]
      }
    >
      <div
        className="form-field"
      >
        <label
          className="field-label"
          htmlFor="checkbox"
        >
          <span
            className="required"
          />
          <span
            className="label"
          >
            checkbox
          </span>
        </label>
        <div
          className="field-input"
        >
          <CheckboxGroup
            api={
              Object {
                "addField": [Function],
                "changeField": [Function],
                "getErrors": [Function],
                "getTouched": [Function],
                "getValues": [Function],
                "removeField": [Function],
                "setValue": [Function],
              }
            }
            changeFieldValue={[Function]}
            label="checkbox"
            name="checkbox"
            options={
              Array [
                Object {
                  "label": "apple",
                  "value": "apple",
                },
                Object {
                  "label": "banana",
                  "value": "banana",
                },
              ]
            }
          >
            <div
              className="checkbox-group"
              id="checkbox"
            >
              <label
                className="checkbox"
              >
                <input
                  checked={false}
                  className="checkbox-input"
                  onChange={[Function]}
                  type="checkbox"
                />
                apple
              </label>
              <label
                className="checkbox"
              >
                <input
                  checked={false}
                  className="checkbox-input"
                  onChange={[Function]}
                  type="checkbox"
                />
                banana
              </label>
            </div>
          </CheckboxGroup>
        </div>
      </div>
    </FormField>
    <FormField
      label="select"
      name="select"
      onChange={[Function]}
      options={
        Array [
          Object {
            "label": "cat",
            "value": "cat",
          },
          Object {
            "label": "dog",
            "value": "dog",
          },
        ]
      }
    >
      <div
        className="form-field"
      >
        <label
          className="field-label"
          htmlFor="select"
        >
          <span
            className="required"
          />
          <span
            className="label"
          >
            select
          </span>
        </label>
        <div
          className="field-input"
        >
          <Select
            api={
              Object {
                "addField": [Function],
                "changeField": [Function],
                "getErrors": [Function],
                "getTouched": [Function],
                "getValues": [Function],
                "removeField": [Function],
                "setValue": [Function],
              }
            }
            changeFieldValue={[Function]}
            label="select"
            name="select"
            options={
              Array [
                Object {
                  "label": "cat",
                  "value": "cat",
                },
                Object {
                  "label": "dog",
                  "value": "dog",
                },
              ]
            }
          >
            <select
              id="select"
              name="select"
              onChange={[Function]}
              value=""
            >
              <option
                value=""
              >
                 --- 
              </option>
              <option
                value="cat"
              >
                cat
              </option>
              <option
                value="dog"
              >
                dog
              </option>
            </select>
          </Select>
        </div>
      </div>
    </FormField>
  </form>
</Form>
`;

exports[`form component change 2`] = `
<Form
  onSubmit={[Function]}
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <FormField
      label="input"
      name="input"
      onChange={[Function]}
    >
      <div
        className="form-field"
      >
        <label
          className="field-label"
          htmlFor="input"
        >
          <span
            className="required"
          />
          <span
            className="label"
          >
            input
          </span>
        </label>
        <div
          className="field-input"
        >
          <Input
            api={
              Object {
                "addField": [Function],
                "changeField": [Function],
                "getErrors": [Function],
                "getTouched": [Function],
                "getValues": [Function],
                "removeField": [Function],
                "setValue": [Function],
              }
            }
            changeFieldValue={[Function]}
            label="input"
            name="input"
            value="apple"
          >
            <input
              autoComplete="off"
              className="input"
              id="input"
              onChange={[Function]}
              onFocus={[Function]}
              type="text"
              value="apple"
            />
          </Input>
        </div>
        <FormError
          name="input"
        />
      </div>
    </FormField>
    <FormField
      label="radio"
      name="radio"
      onChange={[Function]}
      options={
        Array [
          Object {
            "label": "red",
            "value": "red",
          },
          Object {
            "label": "yellow",
            "value": "yellow",
          },
          Object {
            "label": "blue",
            "value": "blue",
          },
        ]
      }
    >
      <div
        className="form-field"
      >
        <label
          className="field-label"
          htmlFor="radio"
        >
          <span
            className="required"
          />
          <span
            className="label"
          >
            radio
          </span>
        </label>
        <div
          className="field-input"
        >
          <RadioGroup
            api={
              Object {
                "addField": [Function],
                "changeField": [Function],
                "getErrors": [Function],
                "getTouched": [Function],
                "getValues": [Function],
                "removeField": [Function],
                "setValue": [Function],
              }
            }
            changeFieldValue={[Function]}
            label="radio"
            name="radio"
            options={
              Array [
                Object {
                  "label": "red",
                  "value": "red",
                },
                Object {
                  "label": "yellow",
                  "value": "yellow",
                },
                Object {
                  "label": "blue",
                  "value": "blue",
                },
              ]
            }
            value="yellow"
          >
            <div
              className="radio-group"
              id="radio"
            >
              <label
                className="radio"
              >
                <input
                  checked={false}
                  className="radio-input"
                  onChange={[Function]}
                  type="radio"
                />
                red
              </label>
              <label
                className="radio checked"
              >
                <input
                  checked={true}
                  className="radio-input"
                  onChange={[Function]}
                  type="radio"
                />
                yellow
              </label>
              <label
                className="radio"
              >
                <input
                  checked={false}
                  className="radio-input"
                  onChange={[Function]}
                  type="radio"
                />
                blue
              </label>
            </div>
          </RadioGroup>
        </div>
        <FormError
          name="radio"
        />
      </div>
    </FormField>
    <FormField
      label="checkbox"
      name="checkbox"
      onChange={[Function]}
      options={
        Array [
          Object {
            "label": "apple",
            "value": "apple",
          },
          Object {
            "label": "banana",
            "value": "banana",
          },
        ]
      }
    >
      <div
        className="form-field"
      >
        <label
          className="field-label"
          htmlFor="checkbox"
        >
          <span
            className="required"
          />
          <span
            className="label"
          >
            checkbox
          </span>
        </label>
        <div
          className="field-input"
        >
          <CheckboxGroup
            api={
              Object {
                "addField": [Function],
                "changeField": [Function],
                "getErrors": [Function],
                "getTouched": [Function],
                "getValues": [Function],
                "removeField": [Function],
                "setValue": [Function],
              }
            }
            changeFieldValue={[Function]}
            label="checkbox"
            name="checkbox"
            options={
              Array [
                Object {
                  "label": "apple",
                  "value": "apple",
                },
                Object {
                  "label": "banana",
                  "value": "banana",
                },
              ]
            }
            value={Array []}
          >
            <div
              className="checkbox-group"
              id="checkbox"
            >
              <label
                className="checkbox"
              >
                <input
                  checked={false}
                  className="checkbox-input"
                  onChange={[Function]}
                  type="checkbox"
                />
                apple
              </label>
              <label
                className="checkbox"
              >
                <input
                  checked={false}
                  className="checkbox-input"
                  onChange={[Function]}
                  type="checkbox"
                />
                banana
              </label>
            </div>
          </CheckboxGroup>
        </div>
        <FormError
          name="checkbox"
        />
      </div>
    </FormField>
    <FormField
      label="select"
      name="select"
      onChange={[Function]}
      options={
        Array [
          Object {
            "label": "cat",
            "value": "cat",
          },
          Object {
            "label": "dog",
            "value": "dog",
          },
        ]
      }
    >
      <div
        className="form-field"
      >
        <label
          className="field-label"
          htmlFor="select"
        >
          <span
            className="required"
          />
          <span
            className="label"
          >
            select
          </span>
        </label>
        <div
          className="field-input"
        >
          <Select
            api={
              Object {
                "addField": [Function],
                "changeField": [Function],
                "getErrors": [Function],
                "getTouched": [Function],
                "getValues": [Function],
                "removeField": [Function],
                "setValue": [Function],
              }
            }
            changeFieldValue={[Function]}
            label="select"
            name="select"
            options={
              Array [
                Object {
                  "label": "cat",
                  "value": "cat",
                },
                Object {
                  "label": "dog",
                  "value": "dog",
                },
              ]
            }
            value="dog"
          >
            <select
              id="select"
              name="select"
              onChange={[Function]}
              value="dog"
            >
              <option
                value=""
              >
                 --- 
              </option>
              <option
                value="cat"
              >
                cat
              </option>
              <option
                value="dog"
              >
                dog
              </option>
            </select>
          </Select>
        </div>
        <FormError
          name="select"
        />
      </div>
    </FormField>
  </form>
</Form>
`;

exports[`form submit 1`] = `
<Form
  onSubmit={[Function]}
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <FormField
      label="input"
      name="input"
    >
      <div
        className="form-field"
      >
        <label
          className="field-label"
          htmlFor="input"
        >
          <span
            className="required"
          />
          <span
            className="label"
          >
            input
          </span>
        </label>
        <div
          className="field-input"
        >
          <Input
            api={
              Object {
                "addField": [Function],
                "changeField": [Function],
                "getErrors": [Function],
                "getTouched": [Function],
                "getValues": [Function],
                "removeField": [Function],
                "setValue": [Function],
              }
            }
            changeFieldValue={[Function]}
            label="input"
            name="input"
          >
            <input
              autoComplete="off"
              className="input"
              id="input"
              onChange={[Function]}
              onFocus={[Function]}
              type="text"
              value=""
            />
          </Input>
        </div>
      </div>
    </FormField>
    <FormField
      label="radio"
      name="radio"
      options={
        Array [
          Object {
            "label": "red",
            "value": "red",
          },
          Object {
            "label": "yellow",
            "value": "yellow",
          },
          Object {
            "label": "blue",
            "value": "blue",
          },
        ]
      }
    >
      <div
        className="form-field"
      >
        <label
          className="field-label"
          htmlFor="radio"
        >
          <span
            className="required"
          />
          <span
            className="label"
          >
            radio
          </span>
        </label>
        <div
          className="field-input"
        >
          <RadioGroup
            api={
              Object {
                "addField": [Function],
                "changeField": [Function],
                "getErrors": [Function],
                "getTouched": [Function],
                "getValues": [Function],
                "removeField": [Function],
                "setValue": [Function],
              }
            }
            changeFieldValue={[Function]}
            label="radio"
            name="radio"
            options={
              Array [
                Object {
                  "label": "red",
                  "value": "red",
                },
                Object {
                  "label": "yellow",
                  "value": "yellow",
                },
                Object {
                  "label": "blue",
                  "value": "blue",
                },
              ]
            }
          >
            <div
              className="radio-group"
              id="radio"
            >
              <label
                className="radio"
              >
                <input
                  checked={false}
                  className="radio-input"
                  onChange={[Function]}
                  type="radio"
                />
                red
              </label>
              <label
                className="radio"
              >
                <input
                  checked={false}
                  className="radio-input"
                  onChange={[Function]}
                  type="radio"
                />
                yellow
              </label>
              <label
                className="radio"
              >
                <input
                  checked={false}
                  className="radio-input"
                  onChange={[Function]}
                  type="radio"
                />
                blue
              </label>
            </div>
          </RadioGroup>
        </div>
      </div>
    </FormField>
    <FormField
      label="checkbox"
      name="checkbox"
      options={
        Array [
          Object {
            "label": "apple",
            "value": "apple",
          },
          Object {
            "label": "banana",
            "value": "banana",
          },
        ]
      }
    >
      <div
        className="form-field"
      >
        <label
          className="field-label"
          htmlFor="checkbox"
        >
          <span
            className="required"
          />
          <span
            className="label"
          >
            checkbox
          </span>
        </label>
        <div
          className="field-input"
        >
          <CheckboxGroup
            api={
              Object {
                "addField": [Function],
                "changeField": [Function],
                "getErrors": [Function],
                "getTouched": [Function],
                "getValues": [Function],
                "removeField": [Function],
                "setValue": [Function],
              }
            }
            changeFieldValue={[Function]}
            label="checkbox"
            name="checkbox"
            options={
              Array [
                Object {
                  "label": "apple",
                  "value": "apple",
                },
                Object {
                  "label": "banana",
                  "value": "banana",
                },
              ]
            }
          >
            <div
              className="checkbox-group"
              id="checkbox"
            >
              <label
                className="checkbox"
              >
                <input
                  checked={false}
                  className="checkbox-input"
                  onChange={[Function]}
                  type="checkbox"
                />
                apple
              </label>
              <label
                className="checkbox"
              >
                <input
                  checked={false}
                  className="checkbox-input"
                  onChange={[Function]}
                  type="checkbox"
                />
                banana
              </label>
            </div>
          </CheckboxGroup>
        </div>
      </div>
    </FormField>
    <FormField
      label="select"
      name="select"
      options={
        Array [
          Object {
            "label": "cat",
            "value": "cat",
          },
          Object {
            "label": "dog",
            "value": "dog",
          },
        ]
      }
    >
      <div
        className="form-field"
      >
        <label
          className="field-label"
          htmlFor="select"
        >
          <span
            className="required"
          />
          <span
            className="label"
          >
            select
          </span>
        </label>
        <div
          className="field-input"
        >
          <Select
            api={
              Object {
                "addField": [Function],
                "changeField": [Function],
                "getErrors": [Function],
                "getTouched": [Function],
                "getValues": [Function],
                "removeField": [Function],
                "setValue": [Function],
              }
            }
            changeFieldValue={[Function]}
            label="select"
            name="select"
            options={
              Array [
                Object {
                  "label": "cat",
                  "value": "cat",
                },
                Object {
                  "label": "dog",
                  "value": "dog",
                },
              ]
            }
          >
            <select
              id="select"
              name="select"
              onChange={[Function]}
              value=""
            >
              <option
                value=""
              >
                 --- 
              </option>
              <option
                value="cat"
              >
                cat
              </option>
              <option
                value="dog"
              >
                dog
              </option>
            </select>
          </Select>
        </div>
      </div>
    </FormField>
  </form>
</Form>
`;

exports[`form submit 2`] = `
<Form
  onSubmit={[Function]}
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <FormField
      label="input"
      name="input"
    >
      <div
        className="form-field"
      >
        <label
          className="field-label"
          htmlFor="input"
        >
          <span
            className="required"
          />
          <span
            className="label"
          >
            input
          </span>
        </label>
        <div
          className="field-input"
        >
          <Input
            api={
              Object {
                "addField": [Function],
                "changeField": [Function],
                "getErrors": [Function],
                "getTouched": [Function],
                "getValues": [Function],
                "removeField": [Function],
                "setValue": [Function],
              }
            }
            changeFieldValue={[Function]}
            label="input"
            name="input"
            value="apple"
          >
            <input
              autoComplete="off"
              className="input"
              id="input"
              onChange={[Function]}
              onFocus={[Function]}
              type="text"
              value="apple"
            />
          </Input>
        </div>
        <FormError
          name="input"
        />
      </div>
    </FormField>
    <FormField
      label="radio"
      name="radio"
      options={
        Array [
          Object {
            "label": "red",
            "value": "red",
          },
          Object {
            "label": "yellow",
            "value": "yellow",
          },
          Object {
            "label": "blue",
            "value": "blue",
          },
        ]
      }
    >
      <div
        className="form-field"
      >
        <label
          className="field-label"
          htmlFor="radio"
        >
          <span
            className="required"
          />
          <span
            className="label"
          >
            radio
          </span>
        </label>
        <div
          className="field-input"
        >
          <RadioGroup
            api={
              Object {
                "addField": [Function],
                "changeField": [Function],
                "getErrors": [Function],
                "getTouched": [Function],
                "getValues": [Function],
                "removeField": [Function],
                "setValue": [Function],
              }
            }
            changeFieldValue={[Function]}
            label="radio"
            name="radio"
            options={
              Array [
                Object {
                  "label": "red",
                  "value": "red",
                },
                Object {
                  "label": "yellow",
                  "value": "yellow",
                },
                Object {
                  "label": "blue",
                  "value": "blue",
                },
              ]
            }
            value="red"
          >
            <div
              className="radio-group"
              id="radio"
            >
              <label
                className="radio checked"
              >
                <input
                  checked={true}
                  className="radio-input"
                  onChange={[Function]}
                  type="radio"
                />
                red
              </label>
              <label
                className="radio"
              >
                <input
                  checked={false}
                  className="radio-input"
                  onChange={[Function]}
                  type="radio"
                />
                yellow
              </label>
              <label
                className="radio"
              >
                <input
                  checked={false}
                  className="radio-input"
                  onChange={[Function]}
                  type="radio"
                />
                blue
              </label>
            </div>
          </RadioGroup>
        </div>
        <FormError
          name="radio"
        />
      </div>
    </FormField>
    <FormField
      label="checkbox"
      name="checkbox"
      options={
        Array [
          Object {
            "label": "apple",
            "value": "apple",
          },
          Object {
            "label": "banana",
            "value": "banana",
          },
        ]
      }
    >
      <div
        className="form-field"
      >
        <label
          className="field-label"
          htmlFor="checkbox"
        >
          <span
            className="required"
          />
          <span
            className="label"
          >
            checkbox
          </span>
        </label>
        <div
          className="field-input"
        >
          <CheckboxGroup
            api={
              Object {
                "addField": [Function],
                "changeField": [Function],
                "getErrors": [Function],
                "getTouched": [Function],
                "getValues": [Function],
                "removeField": [Function],
                "setValue": [Function],
              }
            }
            changeFieldValue={[Function]}
            label="checkbox"
            name="checkbox"
            options={
              Array [
                Object {
                  "label": "apple",
                  "value": "apple",
                },
                Object {
                  "label": "banana",
                  "value": "banana",
                },
              ]
            }
            value={
              Array [
                "apple",
              ]
            }
          >
            <div
              className="checkbox-group"
              id="checkbox"
            >
              <label
                className="checkbox checked"
              >
                <input
                  checked={true}
                  className="checkbox-input"
                  onChange={[Function]}
                  type="checkbox"
                />
                apple
              </label>
              <label
                className="checkbox"
              >
                <input
                  checked={false}
                  className="checkbox-input"
                  onChange={[Function]}
                  type="checkbox"
                />
                banana
              </label>
            </div>
          </CheckboxGroup>
        </div>
        <FormError
          name="checkbox"
        />
      </div>
    </FormField>
    <FormField
      label="select"
      name="select"
      options={
        Array [
          Object {
            "label": "cat",
            "value": "cat",
          },
          Object {
            "label": "dog",
            "value": "dog",
          },
        ]
      }
    >
      <div
        className="form-field"
      >
        <label
          className="field-label"
          htmlFor="select"
        >
          <span
            className="required"
          />
          <span
            className="label"
          >
            select
          </span>
        </label>
        <div
          className="field-input"
        >
          <Select
            api={
              Object {
                "addField": [Function],
                "changeField": [Function],
                "getErrors": [Function],
                "getTouched": [Function],
                "getValues": [Function],
                "removeField": [Function],
                "setValue": [Function],
              }
            }
            changeFieldValue={[Function]}
            label="select"
            name="select"
            options={
              Array [
                Object {
                  "label": "cat",
                  "value": "cat",
                },
                Object {
                  "label": "dog",
                  "value": "dog",
                },
              ]
            }
            value="cat"
          >
            <select
              id="select"
              name="select"
              onChange={[Function]}
              value="cat"
            >
              <option
                value=""
              >
                 --- 
              </option>
              <option
                value="cat"
              >
                cat
              </option>
              <option
                value="dog"
              >
                dog
              </option>
            </select>
          </Select>
        </div>
        <FormError
          name="select"
        />
      </div>
    </FormField>
  </form>
</Form>
`;
